### ISSUE TITLE
Compiler considers a tag in an constant array default argument as a variable

### ISSUE THREAD

**Issue by YashasSamaga** [Zeex/pawn#145]
**Triggers an error**
```
myfunction(a:blabla[3] = blabla:{1, 2, 3})
{

}
```
`error 008: must be a constant expression; assumed zero`

**Does not trigger an error**
```
myfunction(a:blabla[3] = {1, 2, 3})
{

}
```

**Comment by Y-Less** [comment]
A better question might be - is the syntax you used the bug?  This doesn't work:

```pawn
new Float:a[5] = Float:{0, 0, 0, 0, 0};
```

Why should it anywhere else?

**Comment by YashasSamaga** [comment]
It works while calling functions:

```
func(TAG:arr[])
{

}

main ()
{
	func(TAG:{1, 2, 3});
}
```

**Comment by Y-Less** [comment]
Yes, I saw you mention that in the first post.  But several other things work in that context but not others:

```pawn
new x = A:B:C:5; // Error.
Func(A:B:C:x); // Fine.
```

**Comment by YashasSamaga** [comment]
For the record, the fix for the second part of the issue is:
- replace ` if (!matchtag(tags[0],arg->defvalue_tag,TRUE)) with the `if(!checktag(tags, numtags, arg->defvalue_tag))` 

The complete fix depends on the status of #233.

**Comment by Southclaws** [comment]
Is this still a bug? #233 was closed a while ago but I lost track of where this problem was at.
